demo/src/Types.elm

```diff
@@ -54,6 +56,33 @@ to the program as `endpoint` (see above).
 type Msg
     = Endpoint
     | GotQuotes (Result Http.Error (List Quote))
+    | RandomNumber Int
+
+
+{-| Your custom interop type.
+
+Should enumerate the interop (JavaScript) functions which can be called.
+
+-}
+type Interop
+    = GetRandom Int
+
+
+interopEncode : Interop -> Encode.Value
+interopEncode interop =
+    case interop of
+        GetRandom upper ->
+            Encode.int upper
```

Comment:
Note that signatures are not restricted to the usual values that can pass through a port, because we JSON encode the arguments before passing them through the port. This is to avoid a type explosion. The user provides a function to JSON encode arguments to `Interop` functions